sources/packages/api/src/route-controllers/institution-locations/institution-location.institutions.controller.ts

```diff
@@ -112,27 +120,46 @@ export class InstitutionLocationInstitutionsController extends BaseController {
    * Get all active application of a location in an institution
    * with application_status is completed.
    * @param locationId location id.
+   * @param searchCriteria Search text to search active applications.
+   * @param sortField Field to sort active applications.
+   * @param page Current page of paginated result.
+   * @param pageLimit Records per page in a paginated result.
+   * @param sortOrder Sort order of active applications.
+   * @param applicationStatus View status of applications requested by user.
    * @returns Student active application list of an institution location.
    */
   @HasLocationAccess("locationId")
   @Get(":locationId/active-applications")
   async getActiveApplications(
     @Param("locationId") locationId: number,
-  ): Promise<ActiveApplicationSummaryAPIOutDTO[]> {
+    @Query() pagination: ApplicationPaginationOptionsAPIInDTO,
+    @Query("applicationStatus") applicationStatus: ApplicationStatus,
+  ): Promise<PaginatedResults<ActiveApplicationSummaryAPIOutDTO>> {
     const applications = await this.applicationService.getActiveApplications(
       locationId,
+      pagination,
+      applicationStatus,
     );
-    return applications.map((eachApplication) => {
-      const offering = eachApplication.currentAssessment?.offering;
-      return {
-        applicationNumber: eachApplication.applicationNumber,
-        applicationId: eachApplication.id,
-        studyStartPeriod: getISODateOnlyString(offering?.studyStartDate),
-        studyEndPeriod: getISODateOnlyString(offering?.studyEndDate),
-        applicationStatus: eachApplication.applicationStatus,
-        fullName: getUserFullName(eachApplication.student.user),
-      };
-    });
+
+    return {
+      results: applications.results.map((eachApplication) => {
+        const offering = eachApplication.currentAssessment?.offering;
+        return {
+          applicationNumber: eachApplication.applicationNumber,
+          applicationId: eachApplication.id,
+          studyStartPeriod: getISODateOnlyString(offering?.studyStartDate),
+          studyEndPeriod: getISODateOnlyString(offering?.studyEndDate),
+          applicationStatus: this.applicationService.getApplicationStatus(
```

Comment:
Please let's avoid reusing the application status here. It would be better to have another status to represent this.

Reply:
I will make the change.